### DISCUSSION TITLE
Renovate can't access private terraform registry in gitlab.com

### DISCUSSION THREAD

**Discussion by muffl0n** [renovatebot/renovate#21079] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.17.0

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitLab Enterprise Edition 15.11.0-pre 9f7fe3ad3f2

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Detect versions of module in private terraform registry.

### What you tried so far.

Set environment variables
```
  RENOVATE_CUSTOM_ENV_VARIABLES: '{"TF_TOKEN_gitlab_com":"$CI_JOB_TOKEN", "TF_CLI_CONFIG_FILE":"$CI_PROJECT_DIR/.terraformrc", "TF_LOG":"DEBUG"}'
  TF_CLI_CONFIG_FILE: $CI_PROJECT_DIR/.terraformrc
```
And write token to `$CI_PROJECT_DIR/.terraformrc`:
```
before_script:
  - echo -e "credentials \"$CI_SERVER_HOST\" {\n  token = \"$CI_JOB_TOKEN\"\n}" > $TF_CLI_CONFIG_FILE
```
Setting `TF_LOG` to `DEBUG` does not help, too.

In the .gitlab-ci.yml of my repository, I set this and it works like a charm for Terraform:
```
variables:
  TF_CLI_CONFIG_FILE: $CI_PROJECT_DIR/.terraformrc

before_script:
  - echo -e "credentials \"$CI_SERVER_HOST\" {\n  token = \"$CI_JOB_TOKEN\"\n}" > $TF_CLI_CONFIG_FILE
```

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: GET https://gitlab.com/api/v4/packages/terraform/modules/v1/<redacted>/workload-identity-federation/google/versions = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=401 retryCount=0, duration=121) (repository=my/repo)
DEBUG: Datasource unauthorized (repository=my/repo)
       "datasource": "terraform-module",
       "packageName": "gitlab.com/<redacted>/workload-identity-federation/google",
       "url": "https://gitlab.com/api/v4/packages/terraform/modules/v1/<redacted>/workload-identity-federation/google/versions"
DEBUG: Failed to look up terraform-module package gitlab.com/<redacted>/workload-identity-federation/google (repository=my/repo, packageFile=project_prodis.tf, dependency=gitlab.com/<redacted>/workload-identity-federation/google)
```

</details>

**Comment by secustor** [comment, marked as answer]
We do not use the Terraform binary for retrieving versions from registries. 
You have to create a hostrule to supply auth tokens. https://docs.renovatebot.com/configuration-options/#hostrules
